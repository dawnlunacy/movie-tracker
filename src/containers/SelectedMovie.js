import React from 'react';


export const SelectedMovie = ({ title, overview, vote_average, release_date }) => {
  return (
      <main>
        <h1>in selectedmovie</h1>
        {/* <img className="poster-img" alt="Movie Poster" src = {`https://image.tmdb.org/t/p/original/${poster_path}`}/>
        <section className="selected-movie-info">
          <h4>{title}</h4>
          <p>Synopsis:{overview}</p>
          <p>Rating: {vote_average}/10</p>
          <p>Released: {release_date}</p> */}
        {/* </section> */}
      </main>
  )
}