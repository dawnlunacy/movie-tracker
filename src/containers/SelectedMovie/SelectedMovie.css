.selectedMovie {
  color:white;
}

.movie-text {
  cursor: pointer;
}

.selected-movie-container {
  display: flex;
  padding: 0 68px 68px 60px;
  position: absolute;
  top: 260px;
}

.selected-title {
  font-size: 50px;
}

.selected-img {
  width: 100%;
}

.selected-movie-info {
  display: flex;
  flex-direction: column;
  max-height: 400px;
  padding-left: 25px;
}

.selected-synopsis {
  margin: auto;
  width: 50%;
}