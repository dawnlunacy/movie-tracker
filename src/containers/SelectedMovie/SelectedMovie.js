import React from 'react';
import './SelectedMovie.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


export const SelectedMovie = ({ selectedMovie }) => {
  return (
      <main className="selected-movie-container">
        <div className="selected-img-container">
          <img className="selected-img" alt="Movie Poster" src={`https://image.tmdb.org/t/p/original/${selectedMovie.poster_path}`}/>
        </div>
        <section className="selected-movie-info">
          <h2 className="selected-title">{selectedMovie.title}</h2>
          <div className="selected-synopsis">
            <h2>Synopsis</h2>
            <p>{selectedMovie.overview}</p>
          </div>
          <div>
            <h2>Rating</h2>
            <p>{selectedMovie.vote_average}/10</p>
          </div>
          <div>
            <h2>Release Date</h2>
            <p>{selectedMovie.release_date}</p>
          </div>
         </section>
      </main>
  )
}

export const mapStateToProps = state => ({
  selectedMovie: state.selectedMovie
});

export default connect(mapStateToProps, null)(SelectedMovie);

SelectedMovie.propTypes = {
  selectedMovie: PropTypes.object.isRequired
}