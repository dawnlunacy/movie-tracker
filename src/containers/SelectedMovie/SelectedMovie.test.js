import React from 'react';
import { shallow } from 'enzyme';
import { SelectedMovie, mapStateToProps } from './SelectedMovie';

describe('SelectedMovie', () => {

  it('should match snapshot', () => {
    const mockSelectedMovie =           
      {
        poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
        title: "Joker",
        overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
        vote_average: 8.6,
        release_date: 19-20-10,
      }
    const wrapper = shallow(
      <SelectedMovie selectedMovie={mockSelectedMovie}/>
    )

    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return an object with the selectedMovie', () => {
      const mockSelectedMovie =           
      {
        poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
        title: "Joker",
        overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
        vote_average: 8.6,
        release_date: 19-20-10,
      }

      const mockState = {
        selectedMovie: [{selectedMovie: mockSelectedMovie}],
        filter: 'SELECT_MOVIE'
      };
      const expected = {
        selectedMovie: [{selectedMovie: mockSelectedMovie}]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
})