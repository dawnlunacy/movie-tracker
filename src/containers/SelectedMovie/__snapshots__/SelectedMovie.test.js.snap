// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMovie should match snapshot 1`] = `
<main
  className="selected-movie-container"
>
  <div
    className="selected-img-container"
  >
    <img
      alt="Movie Poster"
      className="selected-img"
      src="https://image.tmdb.org/t/p/original//udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
    />
  </div>
  <section
    className="selected-movie-info"
  >
    <h2
      className="selected-title"
    >
      Joker
    </h2>
    <div
      className="selected-synopsis"
    >
      <h2>
        Synopsis
      </h2>
      <p>
        During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.
      </p>
    </div>
    <div>
      <h2>
        Rating
      </h2>
      <p>
        8.6
        /10
      </p>
    </div>
    <div>
      <h2>
        Release Date
      </h2>
      <p>
        -11
      </p>
    </div>
  </section>
</main>
`;
